File: model/src/main/java/cm/aptoide/pt/model/v7/BaseV7Response.java
/*
 * Copyright (c) 2016.
 * Modified by Neurophobic Animal on 22/05/2016.
 */

package cm.aptoide.pt.model.v7;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Created by neuro on 20-04-2016.
 */
@Data
public class BaseV7Response {

	private Info info;
	private List<Error> errors;

	public Error getError() {
		if (errors.size() > 0) {
			return errors.get(0);
		} else {
			return null;
		}
	}

	@Data
	@NoArgsConstructor
	@AllArgsConstructor
	public static class Info {

		private Status status;
		private Time time;

		public enum Status {
			OK, QUEUED, FAIL
		}

		@Data
		private static class Time {

			private double seconds;
			private String human;
		}
	}

	@Data
	public static class Error {

		private String code;
		private String description;
	}
}

File: feature_apps/src/main/java/cm/aptoide/pt/feature_apps/data/network/model/BaseV7Response.kt
package cm.aptoide.pt.feature_apps.data.network.model

open class BaseV7Response {
  var info: Info? = null
  var errors: List<Error>? = null
  val error: Error?
    get() = if (errors != null && errors!!.isNotEmpty()) {
      errors!![0]
    } else {
      null
    }
  val isOk: Boolean
    get() = info != null && info!!.status == Info.Status.OK

  class Info(var status: Status?, var time: Time?) {

    enum class Status {
      OK, QUEUED, FAIL, Processing
    }

    class Time {
      var seconds = 0.0
      var human: String? = null
    }
  }

  class Error {
    var code: String? = null
    var description: String? = null
  }
}